.github/workflows/testing.yml

```diff
@@ -0,0 +1,87 @@
+ name: Continuous Integration
+ 
+ on:
+   push:
+     branches:
+     - '*'
+   pull_request:
+     branches:
+     - '*'
+   schedule:
+     - cron: '59 23 * * *'
+   workflow_dispatch:
+    inputs:
+      version:
+        description: Manual CI Run
+        default: test
+        required: false
+
+ jobs:
+   tests:
+     name: ${{ matrix.os }}, ${{ matrix.environment-file }}
+     runs-on: ${{ matrix.os }}
+     timeout-minutes: 30
+     strategy:
+       matrix:
+         os: [ubuntu-latest]
+         environment-file: [
+            ci/310-oldest.yaml,
+            ci/310-latest.yaml,
+            ci/311-latest.yaml,
+            ci/312-latest.yaml,
+            ci/312-dev.yaml,
+         ]
+         include:
+           - environment-file: ci/312-latest.yaml
+             os: macos-latest
+           - environment-file: ci/312-latest.yaml
+             os: macos-14 # Apple Silicon
+           - environment-file: ci/312-latest.yaml
+             os: windows-latest
+       fail-fast: false
+
+     defaults:
+       run:
+         shell: bash -l {0}
+
+     steps:
+       - name: checkout repo
+         uses: actions/checkout@v4
+         with:
+           fetch-depth: 0 # Fetch all history for all branches and tags.
+       
+       - name: setup micromamba
+         uses: mamba-org/setup-micromamba@v1
+         with:
+           environment-file: ${{ matrix.environment-file }}
+           micromamba-version: 'latest'
+        
+       ##############################################################################
+       #- name: install pulp via pip
+       #  run: |
+       #    pip install pulp
+       #    pulptest
+       ##############################################################################
+       
+       - name: environment info
+         run: |
+           micromamba info
+           micromamba list
+       
+       - name: spatial versions
+         run: 'python -c "import geopandas; geopandas.show_versions();"'
+       
+       - name: run tests
+         run: |
+           pytest pysal \
+           -v \
+           -r a \
+           -n logical \
```

Comment:
In order for multicore testing to work, we need to add `pytest-xdist` to all `ci/` environments.